lib/trino-parquet/src/main/java/io/trino/parquet/reader/TrinoColumnIndexStore.java

```diff
@@ -148,7 +148,9 @@ private static <T> Map<ColumnPath, T> loadIndexes(
             return indexMetadata.stream()
                     .collect(toImmutableMap(
                             ColumnIndexMetadata::getPath,
-                            column -> deserializer.apply(getOnlyElement(chunkReaders.get(column.getPath())).readUnchecked(), column)));
+                            column -> deserializer.apply(
+                                    new ChunkedInputStream(chunkReaders.get(column.getPath()), newSimpleAggregatedMemoryContext()),
```

Comment:
We can convert `chunkReaders` to `Map<ColumnPath, ChunkedInputStream>` before the try.
In the finally we can call `close` on all the map values.